File: tests/07_phases_ompss_2.dg/cxx/success_release_01.cpp
/*
<testinfo>
test_generator="config/mercurium-ompss-2"
test_ignore=yes
test_ignore_reason="feature not supported by nanos6"
</testinfo>
*/
#include<assert.h>

int main()
{
    int x = 1;
    #pragma oss task inout(x)
    {
        x++;
        #pragma oss release out(x)
        assert(x == 2);
    }
    #pragma oss task in(x)
    assert(x == 2);

    #pragma oss taskwait
}

File: tests/07_phases_ompss_2.dg/fortran/success_release_01.f90
! <testinfo>
! test_generator=config/mercurium-ompss-2
! test_ignore=yes
! test_ignore_reason="feature not supported by nanos6"
! </testinfo>
PROGRAM P
IMPLICIT NONE
INTEGER :: X

X = 1
!$OSS TASK INOUT(X)
    X = X + 1
    !$OSS RELEASE OUT(X)
    IF (X /= 2) STOP 1
!$OSS END TASK

!$OSS TASK IN(X)
    IF (X /= 2) STOP 1
!$OSS END TASK

!$OSS TASKWAIT
END PROGRAM P
